### PR TITLE
Protect against null cues

### PR DESCRIPTION
**Pull request by mjgiarlo**

### CODE REVIEW

File: app/javascript/controllers/transcript_controller.js [hunk 1]
```diff
@@ -26,17 +26,13 @@ export default class extends Controller {
     const tracks = this.player.textTracks_?.tracks_
     if (!tracks) return []
 
-    return tracks.filter(track => track.kind === 'captions')
+    return tracks.filter(track => track.kind === 'captions' && track?.cues?.cues_)
```
Comment (mjgiarlo) [inline app/javascript/controllers/transcript_controller.js:29]: Filter out caption tracks without cues *early* so we don't get `null` or `undefined` values later.

File: app/javascript/controllers/transcript_controller.js [hunk 2]
```diff
@@ -26,17 +26,13 @@ export default class extends Controller {
...
   }
 
   get cuesByLanguage() {
     const cues = {}
-    if (this.captionTracks.length == 0) return cues
```
Comment (mjgiarlo) [inline app/javascript/controllers/transcript_controller.js:34]: `this.captionTracks` always returns an array, and we can safely `forEach` an empty array below, so I don't see a need for this check.

File: app/javascript/controllers/transcript_controller.js [hunk 3]
```diff
@@ -26,17 +26,13 @@ export default class extends Controller {
...
 
     this.captionTracks.forEach(track => {
-      if (!track.cues)
-        return
```
Comment (mjgiarlo) [inline app/javascript/controllers/transcript_controller.js:38]: We already filtered these out earlier.

File: app/javascript/controllers/transcript_controller.js [hunk 4]
```diff
@@ -68,8 +64,6 @@ export default class extends Controller {
   }
 
   setupTranscriptLanguageSwitching() {
-    if (this.captionTracks.length == 0) return
```
Comment (mjgiarlo) [inline app/javascript/controllers/transcript_controller.js:71]: `this.captionTracks` always returns an array, and we can safely `forEach` an empty array below, so I don't see a need for this check.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**